### PR TITLE
rewriting network viz, adding more details

### PR DESCRIPTION
**Pull request by bhavyapan**
- worked on refactoring the code in the `network.py` file, made the key `create_network_graph function` easier to understand
- tried out a new library (`graphviz`), rejected it in favor of `plotly`
- fixed the directed graph code - has both colors based on classification of nodes as well as arrows showing directions of transactions now 
- all smoke tests completed 
- ‘micro-level’ viz completed for now 
- planned ‘macro-level’ viz, created a function to collect metrics to test it 
- wrote a short `README.md` doc under `output/network_graphs` for determining the scope of our visualizations as well as further considerations

### CODE REVIEW

**Review by sarahwalker10 — COMMENTED** [review]
Amazing job, Bhavya! I know the graph packages have been giving you a lot of trouble so it's great news that you were able to get one of the graphs working with the colors AND arrows. Few comments/questions:
- Good job commenting throughout your code and creating a README.md, this is great for documentation purposes
- Wanted to flag that the TODO still remains in `src/utils/network.py` to delete all the commented out code (but okay for now since you are still experimenting)
- Out of curiosity, where do `output/network_metrics.txt` and `output/g.gml` come from? Do these files automatically get created whenever you run the network graph and/or are they necessary to support the creation of the network graph?

**Review by trevorspreadbury — CHANGES_REQUESTED** [review]
Great documentation and the graphs provided look like a big step forward! 

A few notes:
- we shouldn't track the output files in git
- I'm not able to run the pipeline. I am getting ```Exception has occurred: NetworkXNotImplemented
not implemented for multigraph type
  File "/project/src/utils/network.py", line 583, in additional_network_metrics
    clustering_coeff = nx.average_clustering(G)
                       ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/project/src/utils/network.py", line 576, in run_network_graph_pipeline
    additional_network_metrics(G)
  File "/project/src/utils/linkage_and_network_pipeline.py", line 221, in clean_data_and_build_network
    run_network_graph_pipeline(2018, 2022, [individuals, organizations, transactions])
  File "/project/scripts/clean_classify_graph_pipeline.py", line 13, in <module>
    clean_data_and_build_network(individuals_table, organizations_table, transactions_table)
networkx.exception.NetworkXNotImplemented: not implemented for multigraph type```

- the linter is failing
Please make sure these issues are fixed for next week to avoid losing any points.

**Comment by bhavyapan** [comment]
> * Out of curiosity, where do `output/network_metrics.txt` and `output/g.gml` come from? Do these files automatically get created whenever you run the network graph and/or are they necessary to support the creation of the network graph?

These files get created whenever we run the network graph pipeline. Based on Trevor's feedback I'm realizing that they do not need to be git tracked.

@trevorspreadbury -- thank you for the feedback. Will address this right away!

**Comment by bhavyapan** [comment]
Pushed a new, cleaner version with passed linter checks (besides the dict/literal issue)!

**Comment by bhavyapan** [comment]
Week 6 Updates 
- Finished cleaning and streamlining viz pipeline code in the network.py file
- Wrote and modified functions required for collecting metrics for cluster visualizations
- Wrote the function plot_macro_level_graph function to plot a simple (non-dynamic) cluster viz using the existing NetworkX Graph and its computed metrics. It is producing some basic results which I hope to build on this week. In order to view it, please run the network.py file in the Docker container from this week's commit by right clicking anywhere on the file and selecting the Run in an Interactive Window option.
- Discussed the file organization between folders with Sarah during Monday's TA session and retained the same

**Review by sarahwalker10 — COMMENTED** [review]
Thank you for thee updates, Bhavya! 
1. You did not pass the precommit because of the "Unnecessary dict() call" - let me know if you have any questions about changing the code from calling dict() to simply building a dictionary (i.e. formatted as {"key": "value", "key": "value", ...} )
2. You may need to run a `git rm output/g.gml` and `git rm output/network_metrics.txt` to remove the already committed output files from GitHub. 

Great job getting started on the macro-level visualization. Looking forward to seeing those improvements next week.

File: src/utils/network.py [hunk 1]
```diff
@@ -1,11 +1,14 @@
-"""Buidling, visualizing, and analyzing networks"""
+"""Buidling, visualizing, and analyzing networks (micro-level)"""
```
Comment (trevorspreadbury) [inline src/utils/network.py:1]: looks like there is macro stuff in here too

Comment (bhavyapan) [reply src/utils/network.py:1]: Oops, will fix this typo in next commit!

**Pull request merged**